src/modules/Packages.rb

```diff
@@ -2420,6 +2355,42 @@ def remote_x11_packages
 
   private
 
+    def fetch_selected(category)
+      items = Pkg.ResolvableProperties("", category, "").select { |i| i["status"] == :selected }
+      items.each_with_object([]) { |i, a| a << i["name"] }.sort
+    end
+
+    # Current package, pattern, product, patch and language selection.
+    #
+    # @return [Hash] selected packages, patterns, products,...
+    def current_proposal
+      { "packages"  => Pkg.GetPackages(:selected, false).compact.sort,
+        "patterns"  => fetch_selected(:pattern),
+        "products"  => fetch_selected(:product),
+        "patches"   => fetch_selected(:patch),
+        "languages" => [Pkg.GetPackageLocale].concat(Pkg.GetAdditionalLocales).compact.sort }
```

Comment:
Switch `[item].concat(array)` to `array.append(item)`:

```suggestion
        "languages" => Pkg.GetAdditionalLocales.append(Pkg.GetPackageLocale).compact.sort }
```

Reply:
If you are calling current_proposal with this change multiple times you will get 
"languages"=>["en_US", "en_US"]} . instead of "languages"=>["en_US"]

Reply:
Thanks to the testcases :-)

Reply:
So do we need appending `uniq`?

Reply:
Ah, duplicates do not matter, if they are always the same...